users-mock/src/androidTest/java/pl/dybczak/githubusers/utils/DisableAnimationsTestRule.kt

```diff
@@ -0,0 +1,46 @@
+package pl.dybczak.githubusers.utils
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import java.io.IOException
+
+
+class DisableAnimationsRule : TestRule {
+
+    override fun apply(base: Statement, description: Description): Statement {
+        return object : Statement() {
+            @Throws(Throwable::class)
+            override fun evaluate() {
+                disableAnimations()
+                try {
+                    base.evaluate()
+                } finally {
+                    enableAnimations()
+                }
+            }
+        }
+    }
+
+    @Throws(IOException::class)
+    private fun disableAnimations() {
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("settings put global transition_animation_scale 0")
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("settings put global window_animation_scale 0")
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("settings put global animator_duration_scale 0")
+    }
+
+    @Throws(IOException::class)
+    private fun enableAnimations() {
```

Comment:
What if phone had animations disabled before test execution? Maybe test rule could read settings before execution, save them and restore afterwards.